### PR TITLE
Light node/insert header chain

### PR DESCRIPTION
**Pull request by jwangdx**
### Description

1. Refactored dpos.verifySeal function.
2. Added verify seal to dpos.verifyHeader and dpos.VerifyHeaders
3. Changed signature InsertHeaderChain for all related interface and implementation. Replace []*types.Header to types.HeaderInsertDataBatch. Change related codes as necessary.
4. Updated blockchain.InsertChain code. verifyHeader called after the last block is inserted.

### CODE REVIEW

File: light/lightchain.go [hunk 1]
```diff
@@ -381,11 +381,12 @@ func (self *LightChain) InsertHeaderChain(chain []*types.Header, checkFreq int)
...
 
 	var events []interface{}
-	whFunc := func(header *types.Header) error {
+	whFunc := func(data types.HeaderInsertData) error {
 		self.mu.Lock()
 		defer self.mu.Unlock()
 
-		status, err := self.hc.WriteHeader(header)
+		header := data.Header
+		status, err := self.hc.WriteHeader(data.Header)
```
Comment (xzhangdx) [inline light/lightchain.go]: already retrieve header from data, so just directly `self.hc.WriteHeader(header)`

File: core/blockchain_insert.go [hunk 2]
```diff
@@ -19,6 +19,8 @@ package core
 import (
 	"time"
 
+	"github.com/DxChainNetwork/godx/consensus"
+
```
Comment (xzhangdx) [inline core/blockchain_insert.go]: format import package: remove empty line between internal packages

File: light/lightchain.go (hunk 1 above)
Comment (jwangdx) [reply light/lightchain.go]: Fixed

File: core/blockchain_insert.go (hunk 2 above)
Comment (jwangdx) [reply core/blockchain_insert.go]: Done.

**Pull request merged**